ftest-toxic-reporter/src/test/java/org/arquillian/cube/q/toxic/reporter/ToxicFunctionalTest.java

```diff
@@ -0,0 +1,105 @@
+package org.arquillian.cube.q.toxic.reporter;
+
+
+import org.apache.commons.io.FileUtils;
+import org.arquillian.cube.HostIp;
+import org.arquillian.cube.impl.util.IOUtil;
+import org.arquillian.cube.q.api.NetworkChaos;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.net.SocketException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.arquillian.cube.q.api.NetworkChaos.DistributedLatencyType.logNormalLatencyInMillis;
+import static org.arquillian.cube.q.api.NetworkChaos.RateType.rate;
+import static org.arquillian.cube.q.api.NetworkChaos.TimeoutType.timeoutInMillis;
+import static org.arquillian.cube.q.api.Q.IterationRunCondition.times;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Arquillian.class)
+public class ToxicFunctionalTest {
+
+    @ArquillianResource
+    private NetworkChaos networkChaos;
+
+    @HostIp
+    private String ip;
+
+    @Rule
+    public TestName name = new TestName();
+
+    private static List<String> executedMethods = new ArrayList<>();
+
+    private static final String LOG_DIR = (System.getProperty("user.dir") + "/target/reports/chaos/");
+
+    @BeforeClass
+    public static void clean_json_files_dir_from_target_reports() throws IOException {
+        File chaosDir = new File(LOG_DIR);
+        if (chaosDir.exists()) {
+            FileUtils.cleanDirectory(new File(LOG_DIR));
+        }
+    }
+
+    @Test(expected = SocketException.class)
```

Comment:
If we can also assert on exception message, to narrow down the possible failure, we could use `@ExpectedException` rule. I guess `SocketException` in this case will have information about connection timed out as a reason, whereas we could also face `Connection refused` if something went wrong. If we capture just exception type we might miss real issue of why the test is failing.